plugins/device/cmd/nvidia/device_test.go

```diff
@@ -0,0 +1,1643 @@
+					"1",
+				},
+				logger: hclog.NewNullLogger(),
+			},
+			ExpectedWriteToChannel: &device.FingerprintResponse{
+				Devices: []*device.DeviceGroup{
+					{
+						Vendor: vendor,
+						Type:   deviceType,
+						Name:   "Name",
+						Devices: []*device.Device{
+							{
+								ID:      "2",
+								Healthy: true,
+							},
+						},
+						Attributes: map[string]string{
+							MemoryMiBAttr:     "10",
+							DriverVersionAttr: "1",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "Check devices are split to multiple device groups",
+			Device: &NvidiaDevice{
+				nvmlCli: &MockNvmlClient{
+					FingerprintResponseReturned: &nvml.FingerprintData{
+						DriverVersion: "1",
+						Devices: []*nvml.FingerprintDeviceData{
+							{
+								UUID:       "1",
+								MemoryMiB:  uint64(10),
+								DeviceName: "Name1",
+							},
+							{
+								UUID:       "2",
+								MemoryMiB:  uint64(11),
+								DeviceName: "Name2",
+							},
+							{
+								UUID:       "3",
+								MemoryMiB:  uint64(12),
+								DeviceName: "Name3",
+							},
+						},
+					},
+				},
+				logger: hclog.NewNullLogger(),
+			},
+			ExpectedWriteToChannel: &device.FingerprintResponse{
+				Devices: []*device.DeviceGroup{
+					{
+						Vendor: vendor,
+						Type:   deviceType,
+						Name:   "Name1",
+						Devices: []*device.Device{
+							{
+								ID:      "1",
+								Healthy: true,
+							},
+						},
+						Attributes: map[string]string{
+							MemoryMiBAttr:     "10",
+							DriverVersionAttr: "1",
+						},
+					},
+					{
+						Vendor: vendor,
+						Type:   deviceType,
+						Name:   "Name2",
+						Devices: []*device.Device{
+							{
+								ID:      "2",
+								Healthy: true,
+							},
+						},
+						Attributes: map[string]string{
+							MemoryMiBAttr:     "11",
+							DriverVersionAttr: "1",
+						},
+					},
+					{
+						Vendor: vendor,
+						Type:   deviceType,
+						Name:   "Name3",
+						Devices: []*device.Device{
+							{
+								ID:      "3",
+								Healthy: true,
+							},
+						},
+						Attributes: map[string]string{
+							MemoryMiBAttr:     "12",
+							DriverVersionAttr: "1",
+						},
+					},
+				},
+			},
+		},
+	} {
+		channel := make(chan *device.FingerprintResponse, 1)
+		testCase.Device.writeFingerprintToChannel(channel)
+		actualResult := <-channel
+		// writeFingerprintToChannel iterates over map keys
+		// and insterts results to an array, so order of elements in output array
+		// may be different
+		// actualResult, expectedResult arrays has to be sorted firsted
+		sort.Slice(actualResult.Devices, func(i, j int) bool {
+			return actualResult.Devices[i].Name < actualResult.Devices[j].Name
+		})
+		sort.Slice(testCase.ExpectedWriteToChannel.Devices, func(i, j int) bool {
+			return testCase.ExpectedWriteToChannel.Devices[i].Name < testCase.ExpectedWriteToChannel.Devices[j].Name
+		})
+		if !reflect.DeepEqual(testCase.ExpectedWriteToChannel, actualResult) {
```

Comment:
We use testify library's require and assert packages elsewhere in Nomad. use `require.Equal` to simplify this test

Reply:
Thanks, that was really helpful!